content/en/developers/integrations/oauth_for_integrations.md

```diff
@@ -124,22 +153,88 @@ You cannot edit a published OAuth client directly, so only go through the publis
 
 For more information about publishing your integration tile and creating your pull request, see the [Marketplace and Integrations documentation][7].
 
-## Further Reading
+## Troubleshooting
+
+### The list of API scopes does not include sending metrics, events, and logs
+
+To send data to Datadog, use the `api_keys_write` scope when generating an API key on behalf of the user. For more information, see [Create an API key](#create-an-api-key).
+
+### API key and token expiration
+
+Refresh tokens do not expire unless the user revokes authorization or the partner revokes the token. If the partner revokes the token, the user must reauthorize the integration to generate new refresh and access tokens. For more information, see the [OAuth2 Authorization Endpoints Reference][13].
+
+### Retrieving API keys
+
+After you create a key using the [api_keys/marketplace][14] endpoint, the key is returned in the response. The key cannot be regenerated or viewed again. Ensure you store the key securely for continuous data transmission. If you lose your API key, follow these steps to revoke and recreate it:
+
```

Comment:
can we add a note under the steps: 
Note: The steps above applies only to partners testing in their own partner sandbox account

Reply:
I added that bit to the section title to make this clearer